monkey/infection_monkey/control.py

```diff
@@ -78,22 +77,12 @@ def send_log(self, log):
             return
         try:
             telemetry = {"monkey_guid": GUID, "log": json.dumps(log)}
-            requests.post(  # noqa: DUO123
-                "https://%s/api/log" % (self.server_address,),
-                data=json.dumps(telemetry),
-                headers={"content-type": "application/json"},
-                verify=False,
-                timeout=MEDIUM_REQUEST_TIMEOUT,
-            )
+            HTTPIslandAPIClient.send_log(self.server_address, json.dumps(telemetry))
```

Comment:
Since we have `IIslandAPIClient`, we should communicate through that interface rather than the concrete implementation.